src/assets/js/main.js

```diff
@@ -85,20 +85,55 @@ function scrollSidebarIntoView() {
   }
 }
 
-function adjustToc() {
-  // Adjustments to the jekyll-toc TOC.
+/**
+ * Adjusts the behavior of the table of contents (TOC) on the page.
+ * 
+ * This function enables a "scrollspy" feature on the TOC, where the active link in the TOC is updated
+ * based on the currently visible section in the page.
+ * 
+ * Enables a "back to top" button in the TOC header.
+ * returns {void}
+ */
+function adjustTOC() {
 
   const tocId = '#site-toc--side';
 
   const tocHeader = document.querySelector(tocId + ' header');
 
   if (tocHeader) {
-    tocHeader.addEventListener('click', function (_) {
-      $('html, body').animate({scrollTop: 0}, 'fast');
+    //Shouldn't it be a "to top" button? 
+    //Its doesn't make sense, it's not intutive.
+    tocHeader.addEventListener('click', (_) => {
+      _scrollToTop();
     });
   }
 
-  $('body').scrollspy({ offset: 100, target: tocId });
+  //Appear not to be used anywhere
+  //'class' active dont make any effect on toc
+  const tocElement = document.querySelector(tocId);
+  if(tocElement){
+    window.addEventListener('scroll', _spyOnScroll);
+  }
+
+  function _scrollToTop() {
+    const distanceBetweenTop = document.documentElement.scrollTop || document.body.scrollTop;
+    if (distanceBetweenTop > 0) {
+      window.requestAnimationFrame(_scrollToTop);
+      const divisorToMakeAnimationFast = 4;
+      window.scrollTo(0, distanceBetweenTop - distanceBetweenTop / divisorToMakeAnimationFast);
+    }
```

Comment:
Cool, this is a clever solution!

However, since the functionality is pretty obscure, what do you think using the browser's built-in smooth scrolling functionality? That way the code is easier to understand.

[According to MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/scrollTo) something like the following should work:

```js
window.scrollTo({
  top: 0,
  behavior: 'smooth'
});
```

If you prefer your current solution, I'm happy with that too :)

Reply:
I don´t used behavior smooth, because I traiedmantain the same velocity that Jquery providing, but I believe that the default behavior is more clean and make the same job in a half of velocity, what is not significantly importantly. I'll apply your suggestion.